### ISSUE TITLE
Icon documentation does not mention how to render the icon sprite on public pages

### ISSUE THREAD

**Issue by coredumperror** [wagtail/wagtail#12665]
The Icon docs go into great detail about how to use Wagtail's Icons, the `{% icon %}` tag, etc, but they seem to be written under the assumption that you'll only ever want to use icons in the admin. Which means they don't have a single mention of the icon sprite, or how to render it on your public-facing pages. 

Attempting to follow the current docs to put icons on your public pages just leads to frustration, because the sprite is *essential*, but never mentioned, so your `{% icon %}` tags simply won't work, for no obvious reason.

### Pertinent section of the Wagtail docs

https://docs.wagtail.org/en/v6.3.1/advanced_topics/icons.html#icon-template-tag

I believe this section is likely the best place to add instructions how to to render the icon sprite, as that would also be template code.

### Details

The icon sprite is rendered like this in the admin templates:

```
  <div data-sprite></div>
  <script src="{% versioned_static 'wagtailadmin/js/icons.js' %}" data-icon-url="{% icon_sprite_url %}"></script>
```

The docs should be changed to include instructions for putting that snippet at the top of the `<body>` tag in your app's base public template, alongside `{% load wagtailadmin_tags %}` (to get the templatetags it uses).

### Working on this

I'd do a PR for this, but I'm not sure exactly how you'd want to express this as documentation.

**Comment by laymonage** [comment]
Thanks, but the sprites are only intended for use in the Wagtail admin, hence it’s in `wagtailadmin_tags` and not `wagtailcore_tags`.

**Comment by coredumperror** [comment]
Well, many people are going to want icons on their public-facing pages, and it'd be extremely easy to let them know how to use the existing system that works fine, as long as the icon sprite is added to the base template.

But if you really want to prevent people from using the icon system outside the admin, the docs should be updated to explicitly state that the `{% icon %}` tag is **only** usable in the admin.

**Comment by SharkyBytes** [comment]
Hi, this issue seems interesting, and I’d like to work on it as my first contribution. Could you please assign it to me?

**Comment by gasman** [comment]
Hi @SharkyBytes - thanks for your interest! You're welcome to work on this, although we still need to reach agreement on what exactly should be done here - whether we should document the process for adding icons to public-facing pages, or add a note stating that they are only usable within the admin - and I would advise holding off on making any code / documentation changes until that's been done.

We don't assign issues outside of the core team - anyone is welcome to pick up any issue.

**Comment by gasman** [comment]
My preference would be to document that this is only usable within the admin. Wagtail has always taken the stance that front-end templating is your own concern, and one that Wagtail stays out of - while we can't and shouldn't stop developers from cherry-picking UI elements from the admin if that's what they want, it's up to the developer themselves to make that work. I don't think we want to start down a path of making Wagtail's UI components available as a formally-supported reusable framework - and if we ever do, we'd certainly want to rework the code to fit that purpose (at the very least, bundling that code snippet into a template tag of its own). Documenting internal details of Wagtail means that we're effectively locked in to doing it that way indefinitely, and unable to change the code to something better without going through a deprecation process.

**Comment by coredumperror** [comment]
There *is* a whole system for adding as many custom icons as you want. My team's project had registered dozens of additional icons via the [wagtail-font-awesome-svg](https://pypi.org/project/wagtail-font-awesome-svg/) library, plus a few custom SVGs we created ourselves.

We had been using a custom template tag that called `wagtail.admin.icons.get_icons()` to put the icon sprite in our templates, but since that's an undocumented method, it actually got renamed in 6.1 (I think), and I had to remediate that during our upgrade, which is what prompted me to go looking for what the Wagtail admin does to get its sprite.

If gasman thinks this shouldn't be an officially supported front-end feature, that's fine. The appropriate resolution to this issue, in that case, would be to make it clear in the docs that the `{% icon %}` tag is exclusively for use in the admin.

**Comment by ravigupta-design** [comment]
> There _is_ a whole system for adding as many custom icons as you want. My team's project had registered dozens of additional icons via the [wagtail-font-awesome-svg](https://pypi.org/project/wagtail-font-awesome-svg/) library, plus a few custom SVGs we created ourselves.
> 
> We had been using a custom template tag that called `wagtail.admin.icons.get_icons()` to put the icon sprite in our templates, but since that's an undocumented method, it actually got renamed in 6.1 (I think), and I had to remediate that during our upgrade, which is what prompted me to go looking for what the Wagtail admin does to get its sprite.
> 
> If gasman thinks this shouldn't be an officially supported front-end feature, that's fine. The appropriate resolution to this issue, in that case, would be to make it clear in the docs that the `{% icon %}` tag is exclusively for use in the admin.

I agree with the point that it should be made clear in documentation that `{% icon %}` tag is exclusively for use in the admin.

I came across this where I wanted to use an icon in my front-end template. The only issue I feel is font-awesome library you linked does not goes with design language of the project UI.

**Comment by thibaudcolas** [comment]
merged the docs upgrades. Suggestions very welcome if people think some of the icons processing should be made reusable outside the CMS. Particularly if it’s a common use case to use the same icons in templates and the CMS. We could document a pattern for this kind of reuse / shared use in our docs. Or in wagtail-font-awesome-svg. Or mention that package in the Wagtail docs.
